client/my-sites/posts/post-controls/index.jsx

```diff
@@ -219,10 +214,43 @@ PostControls.propTypes = {
 	onToggleShare: PropTypes.func,
 	onTrash: PropTypes.func,
 	post: PropTypes.object.isRequired,
+	recordCopyPost: PropTypes.func,
+	recordEditPost: PropTypes.func,
+	recordPreviewPost: PropTypes.func,
+	recordViewPost: PropTypes.func,
+	recordViewPostStats: PropTypes.func,
 	site: PropTypes.object,
 	translate: PropTypes.func,
 };
 
-export default connect( ( state, { site, post } ) => ( {
+const mapStateToProps = ( state, { site, post } ) => ( {
+	canUserDeletePost: canCurrentUser( state, site.ID, 'delete_posts' ),
+	canUserEditPost: canCurrentUser( state, site.ID, 'edit_posts' ),
+	canUserPublishPost: canCurrentUser( state, site.ID, 'publish_posts' ),
 	isPublicizeEnabled: isPublicizeEnabled( state, site.ID, post.type ),
-} ) )( localize( PostControls ) );
+} );
+
+const mapDispatchToProps = dispatch => ( {
+	recordCopyPost: () => dispatch( composeAnalytics(
+		recordGoogleEvent( 'Posts', 'Clicked Copy Post' ),
+		recordTracksEvent( 'calypso_post_controls_copy_post_click' ),
+	) ),
+	recordEditPost: () => dispatch( composeAnalytics(
+		recordGoogleEvent( 'Posts', 'Clicked Edit Post' ),
+		recordTracksEvent( 'calypso_post_controls_edit_post_click' ),
+	) ),
+	recordPreviewPost: () => dispatch( composeAnalytics(
+		recordGoogleEvent( 'Posts', 'Clicked Preview Post' ),
+		recordTracksEvent( 'calypso_post_controls_preview_post_click' ),
+	) ),
+	recordViewPost: () => dispatch( composeAnalytics(
+		recordGoogleEvent( 'Posts', 'Clicked View Post' ),
+		recordTracksEvent( 'calypso_post_controls_view_post_click' ),
+	) ),
+	recordViewPostStats: () => dispatch( composeAnalytics(
+		recordGoogleEvent( 'Posts', 'Clicked View Post Stats' ),
+		recordTracksEvent( 'calypso_post_controls_view_post_stats_click' ),
+	) ),
+} );
```

Comment:
since these are all just creating dispatchable objects, is there a need to create new closures for each one? we could use the shorthand notation here with `react-redux`

```js
const mapDispatchToProps = ( {
	copyPost: composeAnalytics(
		recordGoogleEvent( … ),
		recordTracksEvent( … ),
	),
} );
```